consensus/safety-rules/src/safety_rules.rs

```diff
@@ -264,8 +263,6 @@ impl TSafetyRules for SafetyRules {
 
         self.persistent_storage
             .set_last_voted_round(proposed_block.round())?;
-
-        let validator_signer = self.signer()?;
```

Comment:
Here is a potential bug where signer is not initialized but consensus state is already updated. Fixed in latest commit.

Reply:
how would we get here? this means consensus never initialized safety rules, no?

Reply:
Right. If the safety rule is not initialized (e.g. the key is not found) but this still directly called, it will increment the last voted round, which is definitely not right. I spotted this while writing the specs.

Reply:
ahhh good call